cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -265,4 +326,99 @@ mod tests {
     fn ampersand_but_no_digit() {
         run_comparison("openssl@blah", "Openssl@blah");
     }
+
+    fn run_spdx_comparison(spdx_string: &str, homebrew_dsl: &str) {
+        let result = to_homebrew_license_format(spdx_string).unwrap();
+        assert_eq!(result, homebrew_dsl);
+    }
+
+    #[test]
+    fn spdx_single_license() {
+        run_spdx_comparison("MIT", r#""MIT""#);
+    }
+
+    #[test]
+    fn spdx_single_license_with_plus() {
+        run_spdx_comparison("Apache-2.0+", r#""Apache-2.0+""#);
+    }
+
+    #[test]
+    fn spdx_two_licenses_any() {
+        run_spdx_comparison("MIT OR 0BSD", r#"any_of: ["MIT", "0BSD"]"#);
+    }
+
+    #[test]
+    fn spdx_two_licenses_all() {
+        run_spdx_comparison("MIT AND 0BSD", r#"all_of: ["MIT", "0BSD"]"#);
+    }
+
+    #[test]
+    fn spdx_two_licenses_with_plus() {
+        run_spdx_comparison("MIT OR EPL-1.0+", r#"any_of: ["MIT", "EPL-1.0+"]"#);
+    }
+
+    #[test]
+    fn spdx_three_licenses() {
+        run_spdx_comparison(
+            "MIT OR Apache-2.0 OR CC-BY-4.0",
+            r#"any_of: ["MIT", "Apache-2.0", "CC-BY-4.0"]"#,
+        );
+    }
+
+    #[test]
+    fn spdx_three_licenses_or_and() {
+        run_spdx_comparison(
+            "MIT OR Apache-2.0 AND CC-BY-4.0",
+            r#"any_of: ["MIT", { all_of: ["Apache-2.0", "CC-BY-4.0"] }]"#,
```

Comment:
Looks like Homebrew parses this one to 

```ruby
{:all_of=>[{:any_of=>["MIT", "Apache-2.0"]}, "CC-BY-4.0"]}
```

Reply:
Yeah... I did notice that, but I believe Homebrew is incorrect here 😅. In the [spdx spec](https://spdx.github.io/spdx-spec/v3.0/annexes/SPDX-license-expressions/#d45-order-of-precedence-and-parentheses) they specify that the operator order of precedence is WITH -> AND -> OR, and actually use an example that is basically the same as this test case.

Homebrew is just going left-to-right I believe- it shouldn't be a problem though, for a one-way conversion.

Reply:
Hah! OK, that makes sense. Happy for us to have the more correct version here.